### ISSUE TITLE
can't get performance result on kth 10->20 prediction task

### ISSUE THREAD

**Issue by Liuyu04** [google/e3d_lstm#10]
Hi ,
Thanks for your codes, I have a question about the second experiment on kth action dataset,  the results  I get are very different from yours. 
I train the model from scrach and get:
itr: 200000
training loss: 154.842010
mse per seq: 2290.635904 64.156601 74.118351 81.989762 89.040916 95.132853 100.411171 105.039668 109.337186 113.253830 117.086717 120.570082 124.080130 127.446181 130.632763 133.742619 136.622508 139.171452 141.407400 143.135595 144.260118
psnr per frame: 23.274492 26.751829 25.718252 25.055241 24.532928 24.130718 23.816675 23.556755 23.334009 23.138975 22.954979 22.791883 22.640432 22.499092 22.369574 22.250683 22.139042 22.043234 21.966614 21.913063 21.885820
fmae per frame: 876.817078 700.009094 747.653564 776.031311 800.656433 820.194458 836.558167 850.919067 864.060059 876.001282 887.723083 898.347839 908.817383 918.579346 927.850098 936.855286 945.207214 952.337158 958.406006 963.335754 966.798523
ssim per frame: 0.731547 0.796917 0.778144 0.767673 0.759210 0.751992 0.745644 0.740227 0.735251 0.730660 0.726327 0.722522 0.718926 0.715599 0.712442 0.709682 0.707075 0.705135 0.703583 0.702254 0.701677

And i find that your paper refer to " In the end, we split the database into a **training set of 108,717 sequences** and a **test set of 4,086 sequences**." 
but when i run the code, it shows
begin load datadata/kth_action
there are 127271 pictures
**there are 5200 sequences**
begin load datadata/kth_action
there are 74833 pictures
**there are 3167 sequences**

Why the dataset sequences is different with the paper. Do you know what's wrong ?

**Comment by ManWingloeng** [comment]
> Hi ,
> Thanks for your codes, I have a question about the second experiment on kth action dataset, the results I get are very different from yours.
> I train the model from scrach and get:
> itr: 200000
> training loss: 154.842010
> mse per seq: 2290.635904 64.156601 74.118351 81.989762 89.040916 95.132853 100.411171 105.039668 109.337186 113.253830 117.086717 120.570082 124.080130 127.446181 130.632763 133.742619 136.622508 139.171452 141.407400 143.135595 144.260118
> psnr per frame: 23.274492 26.751829 25.718252 25.055241 24.532928 24.130718 23.816675 23.556755 23.334009 23.138975 22.954979 22.791883 22.640432 22.499092 22.369574 22.250683 22.139042 22.043234 21.966614 21.913063 21.885820
> fmae per frame: 876.817078 700.009094 747.653564 776.031311 800.656433 820.194458 836.558167 850.919067 864.060059 876.001282 887.723083 898.347839 908.817383 918.579346 927.850098 936.855286 945.207214 952.337158 958.406006 963.335754 966.798523
> ssim per frame: 0.731547 0.796917 0.778144 0.767673 0.759210 0.751992 0.745644 0.740227 0.735251 0.730660 0.726327 0.722522 0.718926 0.715599 0.712442 0.709682 0.707075 0.705135 0.703583 0.702254 0.701677
> 
> And i find that your paper refer to " In the end, we split the database into a **training set of 108,717 sequences** and a **test set of 4,086 sequences**."
> but when i run the code, it shows
> begin load datadata/kth_action
> there are 127271 pictures
> **there are 5200 sequences**
> begin load datadata/kth_action
> there are 74833 pictures
> **there are 3167 sequences**
> 
> Why the dataset sequences is different with the paper. Do you know what's wrong ?

Do u know where is the problem? I also found that, and I cannot get the result by their dataset.

**Comment by xiaomingdujin** [comment]
I cannot get the result too.

**Comment by xiaomingdujin** [comment]
Maybe I find the reason, in rnn_cell.py, when calculating the output_gate, new_ global_ memory should be returned, but the code returns  global memory，which is not updated, but even if I  return new_ global_ memory, the result is even worse, so I suspect there is a problem in the transmission of time information

**Comment by roadjiang** [comment]
Duplicated issue. Please refer to the following one for future discussions:
https://github.com/google/e3d_lstm/issues/1
